pipelines/shared/nextflow/color_workflow.nf

```diff
@@ -0,0 +1,202 @@
+}
+
+process get_fasta {
+    publishDir params.final_output_dir, mode: 'copy'
+    input:
+        path id_list_dir
+    output:
+        path 'fasta/'
+    """
+    mkdir fasta
+    mkdir fasta/uniprot_ids fasta/uniref90_ids fasta/uniref50_ids
+    for id_file in $id_list_dir/uniprot_ids/*.txt; do
+        [ -e "\${id_file}" ] || continue
+        fasta_file=\${id_file/%.txt/.fasta}
+        fasta_file=\${fasta_file/#$id_list_dir/fasta}
+        perl $projectDir/../shared/perl/get_sequences.pl --fasta-db ${params.fasta_db} --sequence-ids-file \${id_file} --output-sequence-file \${fasta_file}
+    done
+    for id_file in $id_list_dir/uniref90_ids/*.txt; do
+        [ -e "\${id_file}" ] || continue
+        fasta_file=\${id_file/%.txt/.fasta}
+        fasta_file=\${fasta_file/#$id_list_dir/fasta}
+        perl $projectDir/../shared/perl/get_sequences.pl --fasta-db ${params.fasta_db} --sequence-ids-file \${id_file} --output-sequence-file \${fasta_file}
+    done
+    for id_file in $id_list_dir/uniref50_ids/*.txt; do
+        [ -e "\${id_file}" ] || continue
+        fasta_file=\${id_file/%.txt/.fasta}
+        fasta_file=\${fasta_file/#$id_list_dir/fasta}
+        perl $projectDir/../shared/perl/get_sequences.pl --fasta-db ${params.fasta_db} --sequence-ids-file \${id_file} --output-sequence-file \${fasta_file}
+    done
+    """
+}
+
+process color_ssn {
+    publishDir params.final_output_dir, mode: 'copy'
+    input:
+        path ssn_file
+        path cluster_id_map
+        path cluster_num_map
+    output:
+        path 'ssn_colored.xgmml', emit: 'ssn_output'
+        path 'cluster_colors.txt', emit: 'cluster_colors'
+    """
+    perl $projectDir/../shared/perl/color_xgmml.pl --ssn $ssn_file --color-ssn ssn_colored.xgmml --cluster-map $cluster_id_map \
+        --cluster-num-map $cluster_num_map --cluster-color-map cluster_colors.txt --color-file $projectDir/../shared/perl/colors.tab
+    """
+}
+
+process get_ssn_id_info {
+    input:
+        path ssn_file
+    output:
+        path 'edgelist.txt', emit: 'edgelist'
+        path 'index_seqid_map.txt', emit: 'index_seqid_map'
+        path 'id_index_map.txt', emit: 'id_index_map'
+        path 'seqid_source_map.txt', emit: 'seqid_source_map'
+    """
+    perl $projectDir/../shared/perl/ssn_to_id_list.pl --ssn $ssn_file --edgelist edgelist.txt --index-seqid index_seqid_map.txt \
+        --id-index id_index_map.txt --seqid-source-map seqid_source_map.txt
+    """
+}
+
+process unzip_input {
+    input:
+        path ssn_zipped
+    output:
+        path "ssn____local.xgmml"
+    """
+    perl $projectDir/../shared/perl/unzip_xgmml_file.pl --in $ssn_zipped --out ssn____local.xgmml
+    """
+}
+
+// This is necessary to avoid Docker mounting the same directory twice; this occurs
+// because we need to mount the root of the project so that we can get access to the
+// lib directory. When using a SSN that is inside the project directory then the project
+// directory is also mounted.
```

Comment:
Is this mainly an issue in testing? If so could the tests be restructured to avoid the added complexity

Reply:
I didn't think about it, but yes, this is only an issue when the test data is inside the source code directory tree.  I didn't like my change and I feel it's a hack, so I'll remove this copy stuff and figure out something else for the tests.  The latter change will come in a separate issue/PR.